### ISSUE TITLE
[a11y]: The dropdown component does not have the `role="menu"`

### ISSUE THREAD

**Issue by setupotnis** [carbon-design-system/carbon#16092]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.59.5

### React version

18.2.0

### Automated testing tool and ruleset

IBM equal accessibility checker

### Description

The dropdown has the incorrect role. If I try to set the role for the element, it gets applied to the outer div, but the IBM accessibility checker is checking for the div that gets the aria-label applied to it, which is actually the inner div inside of the dropdown component. 

One other thing is that the options for the dropdown should have a role of `option`.

![image](https://github.com/carbon-design-system/carbon/assets/46113818/c51cc2e9-d51e-4a2d-9878-79befe339e6c)

It's saying that the div that has the `aria-label` should have `role="menu"`. 

### WCAG 2.1 Violation

4.1.2 Name, Role, Value

### Reproduction/example

Prompt Lab

### Steps to reproduce

- Create a project in watsonx
- Create a prompt lab asset
- click the dropdown menu which contains the `save as` option:

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hi, could you reproduce the issue in a [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-18-vite/?file=src%2Findex.scss,src%2FApp.jsx&preset=node=), and post the link back here? It's hard to know exactly what's going wrong without seeing the issue firsthand.
